File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
        if(s.length==0) return 0
        let l=0,r=1,ans=0,map={}
        while(r<s.length){
            while((s[l]==s[r]||s[r-1]==s[r])&&l<r){
                l+=1
            }

            ans=Math.max(r-l,ans)

            r+=1
            let i=l
            while(i<r){
                if(s[i]==s[r]){
                    l=i
                }
                i++
            }






        }
        return ans+1
        
    }
}

File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l,ans=0,0
        maxarea=0
        r=len(heights)-1
        while l<r:

            height=min(heights[l],heights[r])
            width=r-l
            area=height*width
            maxarea=max(maxarea,area)

            if heights[l]<heights[r]:
                l+=1
            else:
                r-=1
        return maxarea
